web/src/app/modules/shared/components/smart/terminal/terminal.component.ts

```diff
@@ -33,57 +30,59 @@ import { Subscription } from 'rxjs';
   templateUrl: './terminal.component.html',
 })
 export class TerminalComponent implements OnDestroy, AfterViewInit {
+  constructor(
+    private terminalService: TerminalOutputService,
+    private wss: WebsocketService,
+    private actionService: ActionService
+  ) {}
+
+  private static sessionID = uuidv4();
   private terminalStream: TerminalOutputStreamer;
   private term: Terminal;
   private fitAddon: FitAddon;
   trackByIdentity = trackByIdentity;
-  sliderServiceSubscription: Subscription;
-
   @Input() view: TerminalView;
   @ViewChild('terminal', { static: true }) terminalDiv: ElementRef;
   @HostListener('click') onClick() {
     this.term.focus();
     this.fitAddon.fit();
   }
 
-  constructor(
-    private terminalService: TerminalOutputService,
-    private wss: WebsocketService,
-    private sliderService: SliderService
-  ) {}
-
-  compareFn(c1: TerminalDetail, c2: TerminalDetail): boolean {
-    return c1 && c2 ? c1.uuid === c2.uuid : c1 === c2;
-  }
-
   ngOnDestroy(): void {
     if (this.terminalStream) {
       this.terminalStream.scrollback.unsubscribe();
       this.terminalStream.line.unsubscribe();
       this.terminalStream = null;
     }
-
-    if (this.sliderServiceSubscription) {
-      this.sliderServiceSubscription.unsubscribe();
-    }
   }
 
   ngAfterViewInit() {
     if (this.view) {
-      const disableStdin = this.view.config.terminal.active ? false : true;
       const logLevel = 'info';
-
+      const { podName, namespace, terminal } = this.view.config;
+      const { active, uuid, command, container } = terminal;
+      const disableStdin = !active;
+      const terminalCommand = {
+        containerName: container,
+        containerCommand: command,
+        name: podName,
+        namespace,
+        kind: 'Pod',
+        apiVersion: 'v1',
+        sessionID: TerminalComponent.sessionID,
+        action: 'overview/commandExec',
+      };
+      this.actionService.perform(terminalCommand);
       this.term = new Terminal({
         logLevel,
         disableStdin,
       });
-
       this.initStream();
       this.enableResize();
       this.term.onData(data => {
-        if (this.view.config.terminal.active === true) {
+        if (active) {
```

Comment:
What should happen if the user exits the terminal from tab?

Reply:
Based on https://github.com/vmware-tanzu/octant/issues/664#issuecomment-593667180, this is enforced on exit:
`If you exit an active session # exit the buffer will remain until you reconnect OR navigate away from the pod.`

Do you think we need to add additional user feedback?

Reply:
Yeah, generalizing this more, there should be a notification that alerts the user if the session is no longer active since there are other failure modes (lost connection, container crash).

Reply:
Added more feedback, so for example if you type `exit` now, terminal will show `process exited...` message